core/src/core2/datalog.clj

```diff
@@ -27,6 +27,21 @@
   (s/or :logic-var ::logic-var
         :aggregate ::aggregate))
 
+(defn- expression-spec [sym spec]
+  (s/and seq?
+         #(= sym (first %))
+         (s/conformer next (fn [v] [sym v]))
+         spec))
+
+(s/def ::not-join
+  (expression-spec 'not-join (s/cat :find ::args-list
+                                    :where (s/+ ::term))))
```

Comment:
Would actually keep the `:args` and `:terms` naming here and rename further down, otherwise it might confuse people when looking at a conformed query.

Reply:
No disagreement from me :+1: